app/models/ability.rb

```diff
@@ -76,4 +84,12 @@ def on_team_for_season?(user, season)
     season && user.roles.student.joins(:team).pluck(:season_id).include?(season.id)
   end
 
+  def supervises?(user, supervisor)
+    is_supervisor = Array.new
+    user.teams.in_current_season.each do |team|
```

Comment:
Since we are only interested if they are a supervisor in at least one team, the entire method body can be rewritten as:

```ruby
user.teams.in_current_season.any? { |team| team.supervisors.include?(supervisor) }
```

(Disclaimer: written from memory, not tested)